packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -60,89 +61,110 @@ def _create_file(
     "--class-name",
     help="Name for the Python Measurement Plug-In Client Class in the generated module.",
 )
+@click.option(
+    "-a",
+    "--all",
+    is_flag=True,
+    help="Creates Python Measurement Plug-In Client for all the active measurement services.",
+)
 @click.option(
     "-o",
     "--directory-out",
     help="Output directory for Measurement Plug-In Client files. Default: '<current_directory>/<module_name>'",
 )
 def create_client(
-    measurement_service_class: str,
+    measurement_service_class: List[str],
     module_name: Optional[str],
     class_name: Optional[str],
+    all: Optional[bool],
     directory_out: Optional[str],
 ) -> None:
     """Generates a Python Measurement Plug-In Client module for the measurement service.
 
     You can use the generated module to interact with the corresponding measurement service.
 
     MEASUREMENT_SERVICE_CLASS: The service class of the measurement.
+    Provide space separated service class to generate multiple clients.
     """
     channel_pool = GrpcChannelPool()
     discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
     built_in_import_modules: List[str] = []
     custom_import_modules: List[str] = []
 
-    if module_name is None or class_name is None:
-        base_service_class = measurement_service_class.split(".")[-1]
-        base_service_class = remove_suffix(base_service_class)
-        if not base_service_class.isidentifier():
+    if all:
+        measurement_service_class = get_available_measurements_service_class(discovery_client)
+    else:
+        if not measurement_service_class:
+            raise click.ClickException("Measurement service class cannot be empty.")
+
+    is_multiple_client_generation = len(measurement_service_class) > 1
+    for service_class in measurement_service_class:
+        if is_multiple_client_generation or module_name is None or class_name is None:
```

Comment:
I don't see the use for this condition. If it is multiple client creation, we don't need to worry about the user provided module name and class name